File: src/main/java/com/lsd/ParticipantType.java
package com.lsd;

import com.lsd.report.model.Participant;
import com.lsd.diagram.ValidComponentName;

public enum ParticipantType {
    ACTOR("actor"),
    BOUNDARY("boundary"),
    COLLECTIONS("collections"),
    CONTROL("control"),
    DATABASE("database"),
    ENTITY("entity"),
    PARTICIPANT("participant"),
    QUEUE("queue");

    private final String type;

    ParticipantType(final String type) {
        this.type = type;
    }

    public Participant called(String name) {
        return new Participant(String.format("%s %s", type, ValidComponentName.of(name)));
    }

    public Participant called(String name, String alias) {
        return new Participant(String.format("%s %s as \"%s\"", type, ValidComponentName.of(name), alias));
    }
}

File: src/main/kotlin/com/lsd/ParticipantType.kt
package com.lsd

import com.lsd.diagram.ValidComponentName
import com.lsd.report.model.Participant

enum class ParticipantType(private val type: String) {
    ACTOR("actor"), 
    BOUNDARY("boundary"), 
    COLLECTIONS("collections"), 
    CONTROL("control"), 
    DATABASE("database"), 
    ENTITY("entity"), 
    PARTICIPANT("participant"),
    QUEUE("queue");

    fun called(name: String): Participant {
        return Participant("$type ${ValidComponentName.of(name)}")
    }

    fun called(name: String, alias: String): Participant {
        return Participant("$type ${ValidComponentName.of(name)} as \"$alias\"")
    }
}